clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -289,11 +289,17 @@ void executeCommandWithRetry(
         return;
       }
       LOGGER.info("Sending command {} of topic: {}, partition: {} to fork process.", command, topicName, partition);
+      if (command.equals(START_CONSUMPTION)) {
+        /**
+         * StartConsumption operation may take long time to wait for non-existence store/version until it times out.
+         * Add version check here so that the long waiting period won't impact forked ingestion process Netty server
+         * performance.
+         */
+        Utils.waitStoreVersionOrThrow(topicName, getStoreIngestionService().getMetadataRepo());
```

Comment:
So main process and forked process will a maintain a different set of metadata repo, so a wait here doesn't mean the forked process will have the same data as the main process.
In theory, this is good enough for the scenario you discovered about the leaked resources.

Reply:
Yes, I acknowledge that, so that's why it is actually not "move" but add additional check in main process.

Reply:
I updated the PR name/description to reflect the change.